src/panels/lovelace/entity-rows/hui-input-text-entity-row.ts

```diff
@@ -0,0 +1,90 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import { PaperInputElement } from "@polymer/paper-input/paper-input";
+import { HassEntity } from "home-assistant-js-websocket";
+
+import "../components/hui-generic-entity-row";
+import "./hui-error-entity-row";
+
+import { HomeAssistant } from "../../../types";
+import { EntityRow, EntityConfig } from "./types";
+
+class HuiInputTextEntityRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+  private _stateObj?: HassEntity;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config) {
+      throw new Error("Configuration error");
+    }
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    this._stateObj = this.hass.states[this._config.entity];
+
+    if (!this._stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
+        <paper-input
+          class="value"
+          no-label-float
+          .value="${this._stateObj.state}"
+          .minlength="${this._stateObj.attributes.min}"
+          .maxlength="${this._stateObj.attributes.max}"
+          .autoValidate="${this._stateObj.attributes.pattern}"
+          .pattern="${this._stateObj.attributes.pattern}"
+          .type="${this._stateObj.attributes.mode}"
+          @change="${this._selectedValueChanged}"
+          placeholder="(empty value)"
+        ></paper-input>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  private get _value(): PaperInputElement {
```

Comment:
This name is a bit weird, as you are returning an input element, not a value. I would call it `inputEl`